ngraph/frontend/onnx_import/src/core/attribute.cpp

```diff
@@ -5,37 +5,49 @@
 #include "core/attribute.hpp"
 #include "core/graph.hpp"
 #include "core/model.hpp"
+#include "ngraph/log.hpp"
 
 namespace ngraph
 {
     namespace onnx_import
     {
-        std::vector<Graph> Attribute::get_graph_array(Model& model) const
-        {
-            std::vector<Graph> result;
-            for (const auto& graph : m_attribute_proto->graphs())
-            {
-                result.emplace_back(graph, model);
-            }
-            return result;
-        }
-
-        Subgraph Attribute::get_subgraph(const Graph& parent_graph) const
+        Subgraph Attribute::get_subgraph(
+            const Graph& parent_graph,
+            const std::map<std::size_t, element::Type_t>& subgraph_inputs_types_map) const
         {
             if (m_attribute_proto->type() != ONNX_NAMESPACE::AttributeProto_AttributeType_GRAPH)
             {
                 throw error::attribute::InvalidData{m_attribute_proto->type()};
             }
 
-            ONNX_NAMESPACE::ModelProto model_proto;
+            auto model_proto = common::make_unique<ONNX_NAMESPACE::ModelProto>();
+
             const auto& graph = m_attribute_proto->g();
-            *(model_proto.mutable_graph()) = graph;
+            model_proto->mutable_graph()->CopyFrom(graph);
 
-            // set opset version and domain from the parent graph
-            *model_proto.mutable_opset_import() = parent_graph.get_opset_imports();
+            const std::size_t subgraph_inputs_count =
+                static_cast<size_t>(model_proto->mutable_graph()->mutable_input()->size());
+            // Use the `subgraph_inputs_types_map` to infer the types for the subgraph inputs
+            for (const auto& input_index_type_map : subgraph_inputs_types_map)
+            {
+                if (input_index_type_map.first >= subgraph_inputs_count)
+                {
+                    NGRAPH_WARN << "Input with index: '" << input_index_type_map.first
+                                << "' was not found in the subgraph for the ONNX Loop operator";
```

Comment:
Is this helper only for ONNX Loop?
If not please rephrase this warning.

Reply:
This is an oversight. Of course, you are right